.github/workflows/ci_cd_release.yml

```diff
@@ -0,0 +1,336 @@
+name: release
+on:
+  push:
+    tags:
+      - "v*.*.*"
+    inputs:
+      PyDynaRunContainer:
+        description: 'Run container name'
+        required: false
+        default: ghcr.io/ansys/pydyna-run:latest
+
+env:
+  MAIN_PYTHON_VERSION: '3.13'
+  PACKAGE_NAME: 'ansys-dyna-core'
+  PACKAGE_NAMESPACE: 'ansys.dyna.core'
+  DOCUMENTATION_CNAME: "dyna.docs.pyansys.com"
+  PYDYNA_RUN_CONTAINER: ${{ github.event.inputs.PyDynaRunContainer || 'ghcr.io/ansys/pydyna-run:latest'}}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  changelog-deployment:
+    name: "Update CHANGELOG (on release)"
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: ansys/actions/doc-deploy-changelog@3e17ebb494f3333996ba36bbbadd47a6ff24ec0e # v9.0.15
+        with:
+          token: ${{ secrets.PYANSYS_CI_BOT_TOKEN }}
+          bot-user: ${{ secrets.PYANSYS_CI_BOT_USERNAME }}
+          bot-email: ${{ secrets.PYANSYS_CI_BOT_EMAIL }}
+
+  code-style:
+    name: "Code style"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ansys/actions/code-style@3e17ebb494f3333996ba36bbbadd47a6ff24ec0e # v9.0.15
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+
+  codegen-testing:
+    name: Build and Testing for codegen
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+
+      - name: Install library
+        run: pip install -e .[codegen]
+
+      - name: Generate codegen
```

Comment:
I don't like that these are duplicated with the PR CI. Can these be factored into yaml templates?

Reply:
yeah sure, we can add that.

Reply:
fixed and added

Reply:
@Revathyvenugopal162 I was thinking all of the jobs that are shared between the nightly and ci pipelines